platforms/core-runtime/build-process-services/src/main/java/org/gradle/api/internal/DefaultClassPathProvider.java

```diff
@@ -29,7 +29,7 @@ public DefaultClassPathProvider(ModuleRegistry moduleRegistry) {
     public ClassPath findClassPath(String name) {
         if (name.equals("GRADLE_RUNTIME")) {
             // Use everything reachable from the daemon implementation
-            return moduleRegistry.getModule("gradle-daemon-main").getAllRequiredModulesClasspath();
+            return moduleRegistry.getModule("gradle-daemon-server").getAllRequiredModulesClasspath();
```

Comment:
"daemon-server" contains the daemon implementation now, "daemon-main" contains only enough to bootstrap the daemon process.